Source/WebCore/rendering/RenderLayerFilters.cpp

```diff
@@ -145,8 +146,13 @@ GraphicsContext* RenderLayerFilters::beginFilterEffect(RenderElement& renderer,
         expandedDirtyRect.expand(flippedOutsets);
     }
 
-    // Calculate targetBoundingBox since it will be used if the filter is created.
-    targetBoundingBox = intersection(filterBoxRect, expandedDirtyRect);
+    if (is<RenderSVGShape>(renderer))
+        targetBoundingBox = enclosingLayoutRect(renderer.objectBoundingBox());
```

Comment:
You could `dynamicDowncast` to get a `RenderSVGShape` reference, and access `currentSVGLayoutRect()` / `frameRectEquivalent()` -- `RenderSVGModelObject` already stores the object bounding box enclosed as `LayoutRect`.

Reference:
```
RenderSVGShape.cpp:        setCurrentSVGLayoutRect(enclosingLayoutRect(m_fillBoundingBox));
```

Reply:
Done. I replaced this else if by 
```
    } else if (auto* shape = dynamicDowncast<RenderSVGShape>(renderer))
        filterRegion = shape->currentSVGLayoutRect();
```